### DISCUSSION TITLE
Colored ESP slice

### DISCUSSION THREAD

**Discussion by bergwerf** [molstar/molstar#1381] [General]
I would like to plot a slice of a volume, in this case with electrostatic potential values, and use a color ramp similar to the picture below:
![image](https://github.com/user-attachments/assets/d63ba978-5288-43be-ad35-9a18cd51ae00)

 I couldn't figure out in the Mol* UI how to do this. Below is an example Mol* file of an electrostatic potential surface.

[molstar_mep_surface.molx.zip](https://github.com/user-attachments/files/18062748/molstar_mep_surface.molx.zip)

![image](https://github.com/user-attachments/assets/d593dfd8-185c-4cb3-a735-a68dc76c64dc)

**Comment by arose** [comment]
That is not quite possible at present. Need two things 1) proper coloring support and 2) support for arbitrary planes (currently only axis aligned planes are supported)

**Reply by bergwerf** [reply]
From my perspective this is just a different variant of an isosurface, where coloring based on a volume is already possible. It would be nice if one could color arbitrary meshes using a volume.

**Reply by papillot** [reply]
Here are the specifications for the JVXL format from Jmol: https://chemapps.stolaf.edu/jmol/docs/misc/JVXL-format.pdf
It seems like it can store a collection of predefined isosurfaces instead of storing volume data.

**Reply by arose** [reply]
> From my perspective this is just a different variant of an isosurface, where coloring based on a volume is already possible. It would be nice if one could color arbitrary meshes using a volume.

we'll get there, our rendering engine already support it but not everything is fully integrated

**Reply by arose** [reply]
FYI, started working on it

![image](https://github.com/user-attachments/assets/25a07115-3fb7-48ea-9d3f-eeb99822642a)
